common/k8s/src/middleware/metadata.rs

```diff
@@ -143,6 +144,10 @@ impl Middleware for K8sMetadata {
                         };
                     }
                 }
+
+                // NOTE: Not sure if having a side effect is desirable here.
+                // Alternative would be to accumulate additional metrics in the LineBufferMut struct.
+                Metrics::fs().increment_lines_with_namespace(&parse_result.pod_namespace);
```

Comment:
Will we be double counting lines now? One increment with the label and one increment against the `global` namespace? 

That's probably not what we want to have happen, because prometheus will automatically aggregate and so if you don't specify a `namespace` label in your promql you'll see double-counted numbers. 

Hmmm. Two options come to mind. Either we set up a new command line flag and use that to change the behavior or we go back to having separate metrics. I don't know which approach would be more accepted upstream. If I were the maintainer, I'd prefer the command line flag, but I have no idea what they'd prefer.

Reply:
Yeah I had thought about that too. There is an existing flag that enables
Kubernetes enrichment, so one way to do this is to switch off the global
increment if that flag is enabled.

There is a problem with that approach because not every file being tailed
is necessarily a container's logs but might be some other system log that
matches the inclusion glob. I think there is a workaround though in the
middleware logic to handle that, but it would require plumbing the config through the `FS` module to disable line counting calls there.

Another possibility is to make the `Metrics::Fs` implementation more explicit when in K8S enrichment mode. So if K8s mode is enabled we just ignore calls to Fs::increment_lines() and add a separate call in the k8s middleware to call `increment_lines_with_namespace` with the default argument (or just expose the Option argument in the interface).